cmd/chain-maind/app/app.go

```diff
@@ -137,7 +137,24 @@ func initRootCmd(rootCmd *cobra.Command, encodingConfig params.EncodingConfig) {
 		if err != nil {
 			return err
 		}
+
+		tmpFilePath := cleanedPath + ".tmp"
+		tmpFile, err := os.Create(tmpFilePath)
+		if err != nil {
+			return err
+		}
+
 		defer func() {
+			if _, e := os.Stat(tmpFilePath); e == nil {
+				if rmErr := os.Remove(tmpFilePath); rmErr != nil {
+					fmt.Printf("Warning remove tmpFile failed!: %s\n", rmErr)
+				}
+			}
+
+			if closeErr := tmpFile.Close(); closeErr != nil {
+				fmt.Printf("Error closing tmpFile: %s\n", closeErr)
+			}
+
 			if closeErr := file.Close(); closeErr != nil {
```

Comment:
I think the `cleanedPath` don't need to be opened now?

Reply:
yes

Reply:
Use WriteFile (github.com/google/renameio) to instead